rust/src/nasl/builtin/cert/mod.rs

```diff
@@ -0,0 +1,415 @@
+    ///                 (e.g. "20280929T143520").
+    ///
+    /// - all Return all available information in a human readable
+    ///          format.  Not yet implemented.
+    ///
+    /// - hostnames Return an array with all hostnames listed in the
+    ///   certificates, i.e. the CN part of the subject and all dns-name
+    ///   type subjectAltNames.
+    ///
+    /// - fpr-sha-256 The SHA-256 fingerprint of the certificate.  The
+    ///                  fingerprint is, as usual, computed over the entire
+    ///                  DER encode certificate.
+    ///
+    /// - fpr-sha-1   The SHA-1 fingerprint of the certificate.  The
+    ///                  fingerprint is, as usual, computed over the entire
+    ///                  DER encode certificate.
+    ///
+    /// - image       Return the entire certificate as binary data.
+    ///
+    /// - algorithm-name  Same as signature-algorithm-name. TODO: Remove it and
+    ///                      leave only signature-algorithm-name.
+    ///
+    /// - signature-algorithm-name  Return the algorithm name used to sign the
+    ///                                certificate. Get the OID of the digest
+    ///                                algorithm and translated to a name from a
+    ///                                list from Wireshark.
+    ///                                See epan/dissectors/packet-pkcs1.c
+    ///
+    /// - public-key-algorithm-name  Return the algorithm name of the public key.
+    ///
+    /// - modulus      Return the RSA public key's modulus found in the
+    ///                   structure of the given cert.
+    ///
+    /// - exponent    Return the RSA public key's exponent found in
+    ///                  the structure of the given cert.
+    ///
+    /// - key-size    Return the size to hold the parameters size in bits.
+    ///                  For RSA the bits returned is the modulus.
+    ///                  For DSA the bits returned are of the public exponent.
+    ///
+    ///
+    /// The following arguments are required:
+    /// - pos(0): Object id of the certificate.
+    ///
+    /// - pos(1): A string with the command to select what to return; see above.
+    ///
+    /// The following arguments are optional:
+    /// - idx Used by certain commands to select the n-th value of a set
+    ///    of values.  If not given 0 is assumed.
+    ///
+    /// A NASL type depending on the used command.
+    #[nasl_function(named(idx))]
+    fn cert_query(
+        &self,
+        cert_handle: usize,
+        query: &str,
+        idx: Option<usize>,
+    ) -> Result<NaslValue, FunctionErrorKind> {
+        let idx = idx.unwrap_or(0);
+        let handle = self.0.read().unwrap();
+
+        let cert = handle.certs.get(&cert_handle).ok_or_else(|| {
+            FunctionErrorKind::WrongArgument("The given file descriptor is not valid.".to_string())
+        })?;
+        let result = match query {
+            "serial" => {
```

Comment:
q: I'm wondering if we need those identifiers later on as well and if so if it wouldn't be better to create an enum with a `TryFrom` impl?